### PR TITLE
I#9 deploy docs only for changes in docs

### LINKED ISSUE #9: Deploy documentation as readable site
Implement mechanism to deploy the `docs` (or similar) folder in the repo to be deployed as a readable navigable documentation site.

### PR DESCRIPTION
**Pull request by shuaibq3**
Resolves #9 

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Github pages deployment is triggered only when `md` files are modified in the `docs` folder.

## How to test
- Create commits and push to this branch to test this PR.
    - Adding `md` or any other file anywhere except in the `docs` folder should not trigger github workflow to deploy docs.
    - Adding/Modifying `md` files anywhere (any hierarchy) inside the `docs` should trigger github workflow to deploy docs. However, it should not trigger for non `md` files.

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [x] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.

## Reviewer Request (optional)
Please make sure to revert the commits by resetting HEAD and force-push to this branch to remove any unnecessary commits and files from his PR.

## Demo Link (optional)
https://craftsmenltd.github.io/BloodConnect/

## Additional Notes (optional)

Once this PR is approved, we will need to remove the additional [branch name](https://github.com/CraftsmenLtd/BloodConnect/pull/31/files#diff-87e860af4b9db6c503ed680b6edb6333c6f8d7659f7d1a779a2487466bbc50f6R8) from the action yml file and create another commit (or preferably force push to keep the history clean). Another approval after that push may be needed to finally merge it to master.

Alternatively, the reviewer can (force) push a followup commit with branch name removed and then approve the PR.

Removal of this branch name (and any additional branch names added) from the github-pages environment is needed after this PR is approved.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

**Comment by melon-ruet** [comment]
Looks good to me
Please remove test branch "I#9-deploy-docs-only-for-changes-in-docs" before merging.

**Comment by melon-ruet** [comment]
Approved

**Review by melon-ruet — APPROVED** [review]
approved

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@ on:
   push:
     branches:
       - master
+      - I#9-deploy-docs-only-for-changes-in-docs
+    paths:
+      - docs/**.md
```
[the remark this answers is not in the document]
Comment (shuaibq3) [reply .github/workflows/deploy-docs.yml:10]: yes tested and found it to cover all subdirectories

**Pull request merged**